torch/optim/adagrad.py

```diff
@@ -104,6 +130,9 @@ def _init_group(self, group, params_with_grad, grads, state_sums, state_steps):
                 state_sums.append(state["sum"])
                 state_steps.append(state["step"])
 
+        is_fused = group["fused"]
+        if is_fused and (has_sparse_grad or has_complex):
+            raise RuntimeError("`fused` does not support sparse grad or complex param")
```

Comment:
Let's move this into the fused function call below for consistency. Also, this would be important to mention in the optimizer documentation for fused kwarg.